src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -775,18 +777,30 @@ public Task FlushAsync(CancellationToken cancellationToken = default(Cancellatio
                 // If return is Task.CompletedTask no awaiting is required
                 if (!ReferenceEquals(initializeTask, Task.CompletedTask))
                 {
-                    return FlushAsyncAwaited(initializeTask, cancellationToken);
+                    return InitializeAndFlushAsyncAwaited(initializeTask, cancellationToken);
                 }
             }
 
-            return Output.FlushAsync(cancellationToken);
+            return FlushAsyncInternal(cancellationToken);
         }
 
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private async Task FlushAsyncAwaited(Task initializeTask, CancellationToken cancellationToken)
+        private async Task InitializeAndFlushAsyncAwaited(Task initializeTask, CancellationToken cancellationToken)
         {
             await initializeTask;
-            await Output.FlushAsync(cancellationToken);
+            await FlushAsyncInternal(cancellationToken);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private async Task FlushAsyncAwaited(CancellationToken cancellationToken)
+        {
+            await FlushAsyncInternal(cancellationToken);
```

Comment:
??? Why not jus put the NoInlining attribute on FlushAsyncInternal and call it everywhere you call FlushAsyncAwaited?